src/Common/Commands.Common.Storage/StorageUtilities.cs

```diff
@@ -26,38 +32,143 @@ public static Uri CreateHttpsEndpoint(string endpointUri)
             return new Uri(endpoint);
         }
 
-        public static CloudStorageAccount GenerateCloudStorageAccount(StorageManagementClient storageClient, string accountName)
+        /// <summary>
+        /// Create a cloud storage account using an ARM storage management client
+        /// </summary>
+        /// <param name="storageClient">The client to use to get storage account details.</param>
+        /// <param name="resourceGroupName">The resource group contining the storage account.</param>
+        /// <param name="accountName">The name of the storage account.</param>
+        /// <returns>A CloudStorageAccount that can be used by windows azure storage library to manipulate objects in the storage account.</returns>
+        public static CloudStorageAccount GenerateCloudStorageAccount(Arm.IStorageManagementClient storageClient,
+            string resourceGroupName, string accountName)
         {
-            var storageServiceResponse = storageClient.StorageAccounts.Get(accountName);
-            var storageKeysResponse = storageClient.StorageAccounts.GetKeys(accountName);
-
-            Uri fileEndpoint = null;
-            Uri blobEndpoint = null;
-            Uri queueEndpoint = null;
-            Uri tableEndpoint = null;
+            if (!TestMockSupport.RunningMocked)
+            {
+                var storageServiceResponse = storageClient.StorageAccounts.GetProperties(resourceGroupName, accountName);
+                Uri blobEndpoint = storageServiceResponse.StorageAccount.PrimaryEndpoints.Blob;
+                Uri queueEndpoint = storageServiceResponse.StorageAccount.PrimaryEndpoints.Queue;
+                Uri tableEndpoint = storageServiceResponse.StorageAccount.PrimaryEndpoints.Table;
+                return new CloudStorageAccount(
+                    GenerateStorageCredentials(storageClient, resourceGroupName, accountName),
+                    blobEndpoint,
+                    queueEndpoint,
+                    tableEndpoint, null);
+            }
+            else
+            {
+                return new CloudStorageAccount(
+                    new StorageCredentials(accountName,
+                        Convert.ToBase64String(Encoding.UTF8.GetBytes(Guid.NewGuid().ToString()))),
+                    new Uri(string.Format("https://{0}.blob.core.windows.net", accountName)),
```

Comment:
Looks scary for me that we have hard-coded endpoints here regardless of the environment. Would propose using AzureEnvironment.BlobSuffixEndpoint instead of the hard-coded value. Feel free to ignore that comment in case I miss some context/assumption that's made here.